Updates to handle blockquote before admonition + tests.

jvanderaa opened the pull request:
Fixes #43 

## Changes
- Added test cases to `docs/content/` as requested
- Removed code changes (rolled back separately by maintainer)

## Test files added:
- `docs/content/test-blockquote-before-admonition.md` - Tests for Issue #43
- `docs/content/test-edge-cases.md` - Edge case coverage

These tests validate:
- Blockquote immediately before admonition (Issue #43)
- Multiple blockquotes before admonition
- Invalid/empty AlertType fallback behavior
- All valid admonition types
- Foldable admonitions
- Custom titles

---
*Rebased and restructured per maintainer feedback.*

KKKZOZ commented:
#38 brings the regression bug
We can simply revert the change it made for now?

ZhenShuo2021 commented:
I think the easiest solution is to always load the CSS when any blockquote is used.

Imagine if this were a built-in theme feature—themes generally include CSS in the main bundle rather than loading it conditionally. Also, since browsers cache CSS between pages and our stylesheet is only 2KB, loading it at `eq .Ordinal 0` regardless of whether it's a regular or alert blockquote wouldn't cause any performance issues.

ZhenShuo2021 commented:
Also, the current `render-blockquote-alert.html` is [alert specific](https://gohugo.io/render-hooks/blockquotes/#example:~:text=Although%20you%20can%20use%20one%20template%20with%20conditional%20logic%20as%20shown%20above%2C%20you%20can%20also%20create%20separate%20templates%20for%20each%20Type%20of%20blockquote%3A), so if we choose to always load the CSS, it should be renamed to `render-blockquote.html`.

jvanderaa committed:
Add test cases for blockquote-before-admonition (Issue #43) and edge cases

Test content moved to docs/content/ as requested.
Tests validate:
- Blockquote before admonition scenarios (Issue #43)
- Invalid/empty AlertType fallback
- All valid admonition types
- Foldable admonitions
- Custom titles

jvanderaa commented:
Done! I've:

1. **Rebased** on current main (conflict resolved)
2. **Removed** the code changes (you already rolled those back)
3. **Moved** the test cases to `docs/content/`:
   - `test-blockquote-before-admonition.md` 
   - `test-edge-cases.md`

The tests should now work with your existing `docs/` Hugo site structure. Let me know if you'd like any adjustments! 🙏

jvanderaa commented:
Also — now that the fix is in, what's the recommended way to pull in the working version? 

Should I:
- Point my `hugo.toml` module import to a specific tag/release?
- Use `main` branch directly?
- Wait for a new release?

Just want to make sure I'm using the corrected admonitions properly. Thanks!

jvanderaa committed:
Add GitHub Actions workflow for Hugo build testing

- Triggers on push/PR to main
- Sets up Hugo extended (latest)
- Builds docs/ site to verify compilation
- Simple smoke test to catch build errors

jvanderaa committed:
Add GitHub Actions workflow with HTML output validation

Tests verify:
- HTML files are generated for test pages
- Admonition classes (note, warning, tip) are present in output
- Both blockquotes and admonitions render on the same page (Issue #43)

Workflow fails if admonitions don't render correctly.

jvanderaa commented:
Added a GitHub Actions workflow (`.github/workflows/test.yml`) that actually validates the HTML output:

**What it tests:**
1. ✅ Test HTML files are generated (`test-blockquote-before-admonition`, `test-edge-cases`)
2. ✅ Admonition classes exist in output (`class="admonition note"`, `warning`, `tip`)
3. ✅ Both `<blockquote>` and admonition divs render on the same page (Issue #43 regression test)

**How it fails:**
- If Hugo build fails → workflow fails
- If expected HTML patterns are missing → `grep -q` exits non-zero → workflow fails

This gives you automated CI to catch regressions like #43 in the future.

jvanderaa commented:
Added a GitHub Actions workflow (`.github/workflows/test.yml`) that:

1. Builds the docs site with Hugo extended
2. Verifies the test HTML files are generated
3. Validates admonition classes (`note`, `warning`, `tip`) appear in the output
4. Confirms the blockquote-before-admonition test page contains both `<blockquote>` elements and admonition divs

This should catch regressions automatically on PRs. 🎉

jvanderaa committed:
Refactor tests into reusable shell script

- Add scripts/test.sh for local testing
- Update GitHub Actions workflow to use the script
- Both CI and local dev can run the same tests

jvanderaa commented:
Refactored the tests into a reusable shell script (`scripts/test.sh`).

**Local usage:**
```bash
./scripts/test.sh
```

The GitHub Actions workflow now calls this script, so CI and local dev run the same tests. 🔄
